tee-worker/litentry/core/assertion-build/src/dynamic/contracts/token_holding_amount/erc20/Ada.sol

```diff
@@ -20,16 +20,13 @@ pragma solidity ^0.8.8;
 import "../../libraries/Identities.sol";
 import "../Constants.sol";
 library Ada {
-	function getTokenBscAddress() internal pure returns (string memory) {
-		return "0x3EE2200Efb3400fAbB9AacF31297cBdD1d435D47";
-	}
-
-	function getTokenEthereumAddress() internal pure returns (string memory) {
-		return "";
-	}
-
-	function getTokenName() internal pure returns (string memory) {
-		return "ada";
+	function getTokenAddress(
+		uint32 network
+	) internal pure returns (string memory) {
+		if (network == Web3Networks.Bsc) {
+			return "0x3EE2200Efb3400fAbB9AacF31297cBdD1d435D47";
+		}
+		revert("Unsupported network");
```

Comment:
I have an another option to discuss, if we combine the getTokenAddress and getTokenNetworks as one as below, would it be simpler for future new token add?
```
struct TokenNetwork {
	uint32 id;
	string tokenAddress;
}

library Bean {
	function getTokenRanges() internal pure returns (uint256[] memory) {
		uint256[] memory ranges = new uint256[](5);
		// ...

		return ranges;
	}

	function getTokenNetworks() internal pure returns (TokenNetwork[] memory) {
		TokenNetwork[] memory networks = new TokenNetwork[](2);
		networks[0] = TokenNetwork(Web3Networks.Ethereum, "0x07da81e9a684ab87fad7206b3bc8d0866f48cc7c");
		networks[1] = TokenNetwork(Web3Networks.Bsc, "0xba7b9936a965fac23bb7a8190364fa60622b3cff");

		return networks;
	}
}
```
For brc20 it should be

```
library Btcs {
	function getTokenRanges() internal pure returns (uint256[] memory) {
		uint256[] memory ranges = new uint256[](5);
		// ...

		return ranges;
	}

	function getTokenNetworks() internal pure returns (TokenNetwork[] memory) {
		TokenNetwork[] memory networks = new TokenNetwork[](1);
		networks[0] = TokenNetwork(Web3Networks.BitcoinP2tr);
		return networks;
	}
}
```

Reply:
Already modified and ran the same test cases above, the test still passed.

Reply:
>would it be simpler for future new token add?

The biggest advantage is that when creating tokenInfo, as long as the network is created, the token address will be considered. In the previous way,  it was easy to forget to add one of them.